### DISCUSSION TITLE
Proposal: publish beta/rc versions to GitHub Package Registry

### DISCUSSION THREAD

**Discussion by o-az** [rainbow-me/rainbowkit#1711] [Ideas]
I'm following your https://github.com/rainbow-me/rainbowkit/tree/rainbowkit-wagmi-v2 progress and proposing that you start publishing beta releases on push, it'll help others test out the beta release and provide feedback.

Here's my proposal:

- Use [GitHub Package Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-npm-registry) for non-main releases. Continue using npm for main releases - so your npm publishing flow remains unchanged.
- GitHub Packages work the same as npm, just a different registry. Here's an example of an npm package published to GitHub Package Registry: https://github.com/o-az/typescript-template/pkgs/npm/typescript-template.

I'm happy to pick up this task and work on it e2e.

**Comment by 0xmkh** [comment]
@o-az Thanks for the suggestion!

As of now we haven't thought on publishing the package on push. The reason for this is because currently this is in `beta`. Meaning we're reviewing everything internally before making a release on our end. 

This way we're making sure to not introduce bugs every time we do release on push. Wagmi team also publishes new fixes actively so we're also keeping track of that.

Also `beta` will soon be merged to production and will no longer be needed.
